compiler/src/Language/Bond/Codegen/Cpp/Types_h.hs

```diff
@@ -146,13 +146,13 @@ namespace std
         #{initMetadata}
     };
 
-    #{template}inline void swap(#{structName}& left, #{structName}& right)
+    #{template}inline void swap(#{qualifiedStructName}& left, #{qualifiedStructName}& right)
     {
         left.swap(right);
     }|]
       where
         template = CPP.template s
-        structName = CPP.structName s
+        qualifiedStructName = CPP.qualifiedStructName cpp s
```

Comment:
Another option is to do for `swap` what we already do [here](https://github.com/tstein/bond/blob/9661ff940190b99569d6585af1a2224691336b15/compiler/src/Language/Bond/Codegen/Cpp/Types_h.hs#L106).